app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision/cloning.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'json'
+
+# Provide all configurable aspects of the actual provision operation.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision::Cloning
+  # Called during provision. Prints message to log file.
+  # @param clone_options [Hash] Options used for clone operation.
+  # @return [Boolean] The result of the log message.
+  def log_clone_options(clone_options)
```

Comment:
I'm surprised there isn't a generic version of this method in the base `ManageIQ::Providers::CloudManager::Provision::StateMachine` class, it could look very much like what you have here.

It's not good that the base class calls a method that isn't defined at all, not even one that raises an unimplemented exception

Reply:
Dunno, I took the method from the Power implementation and reworked it.

Reply:
Is this essentially a duplicate of line 22 [[ref]](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/198/files#diff-ce698cd249242f93ff57c4152d96a14261fffb66e561b128da86f060bbc78414R22)

It looks like they both log `JSON.dump(clone_options)`

Reply:
Sort of except that this one always goes to _log and is info. The other goes to logger and is debug. In production they should log the same thing. In development it ensures that all debug logging goes to $ibm_cloud_log. Which means that the developer doesn't have to tail both evm and ibm_cloud to see what is going on.

Reply:
Json dump is so that I can copy the values into a text editor and run a json formatter on it. Much easier to verify and deal with.

Reply:
Yeah I can see that.  Does that mean we'd do this in production too, since it's only a development convenience?